linera-witty/src/runtime/wasmer/function.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementations of [`InstanceWithFunction`] for Wasmer instances.
+
+use super::{
+    parameters::WasmerParameters, results::WasmerResults, EntrypointInstance, ReentrantInstance,
+};
+use crate::{
+    memory_layout::FlatLayout, primitive_types::FlatType, InstanceWithFunction, Runtime,
+    RuntimeError,
+};
+use frunk::{hlist_pat, HList};
+use wasmer::{AsStoreRef, Extern, FromToNativeWasmType, TypedFunction};
+
+/// Implements [`InstanceWithFunction`] for functions with the provided amount of parameters.
+macro_rules! impl_instance_with_function {
+    ($( $names:ident : $types:ident ),*) => {
+        impl_instance_with_function!(@generate(EntrypointInstance) $( $names: $types ),*);
+        impl_instance_with_function!(@generate(ReentrantInstance<'_>) $( $names: $types ),*);
+    };
+
+    (@generate($instance:ty) $( $names:ident : $types:ident ),*) => {
```

Comment:
nit: Seems like you just need another macro instead of using `@generate(..)` ?

Reply:
Done.